packages/@aws-cdk/aws-glue/lib/data-format.ts

```diff
@@ -201,6 +201,13 @@ export interface DataFormatProps {
    * @default - No classification is specified.
    */
   readonly classificationString?: ClassificationString;
+
+  /**
+   * Additional serialization parameters
+   *
+   * @default No dataFormat parameters are specified
+   */
+  readonly serializationParameters?: { [key: string]: any };
```

Comment:
```suggestion
  readonly serializationParameters?: { [key: string]: string };
```

It looks like the value has to be a string. Is that the case?

Reply:
Thank you for your review.

The CfnTable serde parameters is any type, so I guess it would be less strict to have "[key: string]: any" type (instead of "[key: string]: string" type.

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue.CfnTable.StorageDescriptorProperty.html#parameters

in Cloudformation, it is JSON Type : https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-glue-table-serdeinfo.html#cfn-glue-table-serdeinfo-parameters. 

Proposal : I let this one unchanged

Reply:
I revise my previous proposal : It look likes even complex values are stored in string,

see https://docs.amazonaws.cn/en_us/athena/latest/ug/avro-serde.html :
```sql
CREATE EXTERNAL TABLE flights_avro_example (
   yr INT,
   flightdate STRING,
   uniquecarrier STRING,
   airlineid INT,
   carrier STRING,
   flightnum STRING,
   origin STRING,
   dest STRING,
   depdelay INT,
   carrierdelay INT,
   weatherdelay INT
)
PARTITIONED BY (year STRING)
ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe'
WITH SERDEPROPERTIES ('avro.schema.literal'='
{
   "type" : "record",
   "name" : "flights_avro_subset",
   "namespace" : "default",
   "fields" : [ {
      "name" : "yr",
      "type" : [ "null", "int" ],
      "default" : null
   }, {
      "name" : "flightdate",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "uniquecarrier",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "airlineid",
      "type" : [ "null", "int" ],
      "default" : null
   }, {
      "name" : "carrier",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "flightnum",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "origin",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "dest",
      "type" : [ "null", "string" ],
      "default" : null
   }, {
      "name" : "depdelay",
      "type" : [ "null", "int" ],
      "default" : null
   }, {
      "name" : "carrierdelay",
      "type" : [ "null", "int" ],
      "default" : null
   }, {
      "name" : "weatherdelay",
      "type" : [ "null", "int" ],
      "default" : null
    } ]
}
')
STORED AS AVRO
LOCATION 's3://athena-examples-myregion/flight/avro/';
```

Reply:
I'm now wondering if we should change the API to have a better typed experience. It looks like most of the formats have specific serde properties. I'm thinking we may want to change the static properties to methods. Something like: 

```ts
interface CsvOptions {
  /**
   * @default `,`
   */
  seperatorChar?: string;
}
public static csv(props?: CsvOptions): DataFormat;

DataFormat.csv({
  serdeParameters: {
    seperatorChar: ",",
    quoteChar: "'",
    escapeChar: "\\",
  },
});
```

Reply:
+1 to @corymhall's suggestion here.

Reply:
👍  I definitively agree with the better typed experience, and I like your static method proposal. 

But I'm not very confortable with having all possible ser/de parameters keys. :
- Some may be hidden in the different serde implementation (and maybe not very well documented), 
- How about serde evolution/bug fixing that mat introduce new parameter?
- We need to think on how to integrate the more complex options:
     - `ion.<column>.path_extractor` : https://docs.aws.amazon.com/athena/latest/ug/ion-serde-using-ion-serde-properties.htmlhttps://docs.aws.amazon.com/athena/latest/ug/ion-serde-using-ion-serde-properties.html
     - mapping.<column> : https://docs.aws.amazon.com/athena/latest/ug/openx-json-serde.html

Maybe we need to : 
- have some abstraction on the key/value object, that will allow adding adding new "keys" without having to create new issues/PR
- have dedicated classes for configuration, per serde, and will handle `ion.<column>.path_extractor` or other specific logics
- having enum per serde that will helps having the strongly typed experience

Reply:
Proposal :

```typescript
export interface JsonOptions {
  ignoreMalformedJson?: boolean;
  dotsInKeys?: boolean;
  caseInsensitive?: boolean;
  mappings?: { [key: string]: string };
}

export class DataFormat {

    /**
   * Stored as plain text files in JSON format.
   * Uses OpenX Json SerDe for serialization and deseralization.
   *
   * @see https://docs.aws.amazon.com/athena/latest/ug/json.html
   */
  public static json(jsonOptions?: JsonOptions): DataFormat {
    return new JsonDataFormat({
      inputFormat: InputFormat.TEXT,
      outputFormat: OutputFormat.HIVE_IGNORE_KEY_TEXT,
      serializationLibrary: SerializationLibrary.OPENX_JSON,
      classificationString: ClassificationString.JSON,
    }, jsonOptions);
  }

export class JsonDataFormat extends DataFormat {

  public readonly options?: JsonOptions;

  constructor(props: DataFormatProps, options?: JsonOptions) {
    super(props);
    this.options = options;
  }

  toSerializationParameters(): { [p: string]: string } {
    return {
      ...(this.options && this.options.ignoreMalformedJson && { 'ignore.malformed.json': `${this.options.ignoreMalformedJson}` }),
      ...(this.options && this.options.dotsInKeys && { 'dots.in.keys': `${this.options.dotsInKeys}` }),
      ...(this.options && this.options.caseInsensitive && { 'case.insensitive': `${this.options.caseInsensitive}` }),
      ...(this.options && this.options.mappings && this.options && this.options.mappings && Object.entries(this.options?.mappings).reduce((p, [key, value]) => ({ ...p, [`mapping.${key}`]: value }), {} )),
    };
  }
}
```
and usage 

```typescript
  const tableStack = new cdk.Stack(app, 'table');
  new glue.Table(tableStack, 'Table', {
    database,
    tableName: 'table',
    columns: [{
      name: 'col',
      type: glue.Schema.STRING,
    }],
    dataFormat: glue.DataFormat.json({
      caseInsensitive: true,
      mappings: {
        col: 'Col',
      },
    }),
  });
```